src/wp-includes/class-wp-scripts.php

```diff
@@ -281,6 +281,24 @@ public function do_item( $handle, $group = false ) {
 			$intended_strategy = '';
 		}
 
+		// If the script is in the header group, current output is the header, the intended strategy is `defer`, the
+		// actual strategy is not defer and all dependent scripts are in the footer - move this script to the footer.
+		if ( 'defer' === $intended_strategy && 'defer' !== $strategy && 0 === $this->groups[ $handle ] && 0 === $group ) {
+			// Check that all depedencies are in the footer.
+			$all_deps_in_footer = true;
+			foreach ( $this->get_dependents( $handle ) as $dep ) {
+				if ( isset( $this->groups[ $dep ] ) && 1 !== $this->groups[ $dep ] ) {
+					$all_deps_in_footer = false;
+					break;
+				}
+			}
+			if ( $all_deps_in_footer ) {
+				$this->in_footer[] = $handle;
+				$this->groups[ $handle ] = 1;
+				return;
```

Comment:
```suggestion
				return false;
```

Reply:
I already changed this. I also removed the line `$this->groups[ $handle ] = 1;` because the other instance where a script is moved to the footer - https://github.com/WordPress/wordpress-develop/blob/a4f6bccb134b251483777a47259f0593bec00237/src/wp-includes/class-wp-scripts.php#L252-L254